### PR TITLE
Improve logic for get_test_data()

### PR DESCRIPTION
**Pull request by j-smola**
# Description
## What changed?
Related JIRA issue:  GDR--1927
  
## Why was it changed?

Refactor the code in gDRimport to meet reviewer's criteria - function `get_test_data()` returns object class S4 `gdr_test_data`.
  
# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [x] I added documentation for any code changed/added
- [x] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [x] Package version bumped
- [x] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by bczech — APPROVED** [review]
Good job! 💪🏻

File: R/load_files.R [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #' 
 #' @examples
 #'  td <- get_test_data()
-#'  l_tbl <- load_data(td$m_file, td$t_files, td$r_files)
+#'  l_tbl <- load_data(manifest_path(td), template_path(td), result_path(td))
```
Comment (allisonvuong) [inline R/load_files.R:14]: Now that we have the S4 object, we can use it by taking the object itself in as the argument like `load_data(td)`. Then, the `manifest_path`, `template_path`, `result_path`, etc. will be taken from the object

**Review by allisonvuong — COMMENTED** [review]
Janina, speedy work. Please reference the getter functions like `manifest_path(td1)` instead of `td1@manifest_path` so that we do not reference internals. This protects against future changes. Imagine for example, that we rename the slot `manifest_path` to `mpath`. We would then only need to make changes in the `manifest_path` getter to use slot `mpath`, rather than changing all instances where we accessed the slot with `@`.

File: R/load_files.R (hunk 1 above)
Comment (j-smola) [reply R/load_files.R:14]: :)

**Review by marcin-kam — APPROVED** [review]

**Comment by darsoo** [comment]
Please restore `inst/extdata/data_for_unittests/PSets.rds`

**Pull request merged**